### PR TITLE
ASQ 7.4 delayed delivery doco

### PR DESCRIPTION
**Pull request by SeanFeldman**
**Don NOT merge** until release version is on MyGet (pending ASQ [PR](https://github.com/Particular/NServiceBus.AzureStorageQueues/pull/240))

- [x]  New documentation/snippets
- [x] Update samples + smoke test

### CODE REVIEW

**Comment by SeanFeldman** [comment]
@Particular/azure-maintainers please review

**Comment by SeanFeldman** [comment]
Samples update blocked until release version is on myget.

File: transports/azure-storage-queues/delayed-delivery.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+component: ASQ
+tags:
+- Azure
+related:
+- persistence/azure-storage/performance-tuning
+reviewed: 2017-04-24
+---
+
+
+In Versions 4.7 and above, the Azure Storage Queues transport no longer relies on the [timeout manager](/nservicebus/messaging/timeout-manager.md) to implement [delayed delivery](/nservicebus/messaging/delayed-delivery.md). Instead, the transport creates infrastructure using Azure Storage account its provided with for queuing which can delay messages using native Storage features.
```
Comment (SimonCropp) [inline transports/azure-storage-queues/delayed-delivery.md]: > In Versions 4.7 and above

u mean 7.4?

Comment (danielmarbach) [inline transports/azure-storage-queues/delayed-delivery.md]: Right I think this should be 7.4

Comment (danielmarbach) [inline transports/azure-storage-queues/delayed-delivery.md]: > Instead, the transport creates infrastructure using Azure Storage account its provided with for queuing which can delay messages using native Storage features.

is a bit complex to read. How about

> Instead, the transport creates infrastructure for queuing which can delay messages using native Storage features.

Mentioning the storage account is a detail which can be left out for better readability

File: transports/azure-storage-queues/delayed-delivery.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+
+## How it works
+
+When an endpoint is started, the transport creates a storage table and a storage container to work together to provide the necessary infrastructure to support delayed messages. When a message needs to be delayed, it will be stored by the transport in a storage table. To ensure a single copy of delayed messages are dispatched by any endpoint instance, a storage container is used for leasing access to the delayed messages table.
```
Comment (danielmarbach) [inline transports/azure-storage-queues/delayed-delivery.md]: Remove `to work together`

File: transports/azure-storage-queues/delayed-delivery.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+By default, storage table and storage container names are constructed using the following logic: `delayesHASH` where `HASH` is a SHA-1 hash of the endpoint name.
```
Comment (danielmarbach) [inline transports/azure-storage-queues/delayed-delivery.md]: Not sure if it is clear what `delayes` means in this convention description

File: transports/azure-storage-queues/delayed-delivery.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+## How it works
+
+When an endpoint is started, the transport creates a storage table and a storage container to work together to provide the necessary infrastructure to support delayed messages. When a message needs to be delayed, it will be stored by the transport in a storage table. To ensure a single copy of delayed messages are dispatched by any endpoint instance, a storage container is used for leasing access to the delayed messages table.
+
+By default, storage table and storage container names are constructed using the following logic: `delayesHASH` where `HASH` is a SHA-1 hash of the endpoint name. 
+
+
+### Overriding table/container name
+
+Delayed messages table and container can be provided a custom name with Delayed Delivery API:
```
Comment (danielmarbach) [inline transports/azure-storage-queues/delayed-delivery.md]: How about

> Delayed messages table and container names can be overridden with a custom name:

File: transports/azure-storage-queues/delayed-delivery.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+### Disabling the timeout manager
+
+To assist with the upgrade process, the timeout manager is still enabled by default, so any delayed messages already stored in the endpoint's persistence database before the upgrade will be sent when their timeouts expire. Any delayed messages sent after the upgrade will be sent through the delay infrastructure even though the timeout manager is enabled.
+
+Once an endpoint has no more delayed messages in its persistence database, there is no more need for the timeout manager. It can be disabled by calling:
+
+snippet: delayed-delivery-disable-timeoutmanager
+
+At this point, the `-timeouts` and `-timeoutsdispatcher` queues for the endpoint can be deleted from the storage account. In addition, the endpoint no longer requires timeout persistence, so those storage tables can be removed from the persistence database as well.
```
Comment (danielmarbach) [inline transports/azure-storage-queues/delayed-delivery.md]: Should this section link to a persistence doco page?

File: transports/azure-storage-queues/delayed-delivery.md (hunk 1 above)
Comment (SeanFeldman) [inline transports/azure-storage-queues/delayed-delivery.md]: @danielmarbach yet, that will make the text simpler, but will omit an important technical detail which should be documented - same storage account used.

Middle ground suggestion:

> Instead, the transport creates infrastructure for queuing which can delay messages using native Storage features. Storage account provided for queuing is used to implement delayed delivery.

File: transports/azure-storage-queues/delayed-delivery.md (hunk 3 above)
Comment (SeanFeldman) [inline transports/azure-storage-queues/delayed-delivery.md]: Changed that. And fixed the typo 🙂

File: transports/azure-storage-queues/delayed-delivery.md (hunk 5 above)
Comment (SeanFeldman) [inline transports/azure-storage-queues/delayed-delivery.md]: Related link(s) will appear right after the last section (this one).
I've added in addition to the already existing one a general persistence and ASP specific ones

**Comment by SeanFeldman** [comment]
@danielmarbach I've addressed your comments. Please review, but do not merge - still have samples to smoke test and update. Thanks.

**Comment by SeanFeldman** [comment]
@danielmarbach the build is green, comments addressed. Could you please review? Thanks.

**Pull request merged**